File: assignment/A3TaTanPhong_SE18D04_BE/src/main/java/fu/se/sba301/phongtt/a3tatanphong_se18d04/services/BookingService.java
package fu.se.sba301.phongtt.a3tatanphong_se18d04.services;

import fu.se.sba301.phongtt.a3tatanphong_se18d04.dto.request.BookingRequest;
import fu.se.sba301.phongtt.a3tatanphong_se18d04.dto.response.BookingResponse;

import java.util.List;

public interface BookingService {

    BookingResponse createBooking(String email, BookingRequest request);

    List<BookingResponse> getMyBookings(String email);

    List<BookingResponse> getAllBookings();
}

File: assignment/A3TaTanPhong_SE18D04_FE/src/services/bookingService.js
import api from "./api";

export const bookingService = {
  // CUSTOMER: Tạo đơn đặt phòng mới (BookingRequest)
  createBooking: async (bookingRequest) => {
    const response = await api.post("/customer/bookings", bookingRequest);
    return response.data;
  },

  // CUSTOMER: Xem lịch sử đặt phòng cá nhân
  getMyBookings: async () => {
    const response = await api.get("/customer/bookings");
    return response.data;
  },

  // STAFF: Xem toàn bộ đơn đặt phòng của hệ thống
  getAllBookings: async () => {
    const response = await api.get("/staff/bookings");
    return response.data;
  },
};
